streaming/src/main/scala/org/apache/gearpump/streaming/dsl/window/api/WindowFunction.scala

```diff
@@ -32,35 +32,34 @@ object WindowFunction {
   }
 }
 
-trait WindowFunction[T] {
+trait WindowFunction {
 
-  def apply(context: WindowFunction.Context[T]): Array[Window]
+  def apply[T](context: WindowFunction.Context[T]): Array[Window]
 
   def isNonMerging: Boolean
 }
 
-abstract class NonMergingWindowFunction[T] extends WindowFunction[T] {
+abstract class NonMergingWindowFunction extends WindowFunction {
 
   override def isNonMerging: Boolean = true
 }
 
-case class GlobalWindowFunction[T]() extends NonMergingWindowFunction[T] {
+case class GlobalWindowFunction() extends NonMergingWindowFunction {
 
-  override def apply(context: WindowFunction.Context[T]): Array[Window] = {
+  override def apply[T](context: WindowFunction.Context[T]): Array[Window] = {
     Array(Window(Instant.ofEpochMilli(MIN_TIME_MILLIS),
```

Comment:
We can create just one static object, no need to create one each time as it's immutable.